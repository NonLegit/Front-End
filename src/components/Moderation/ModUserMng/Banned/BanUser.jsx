/* eslint-disable import/no-cycle */
/* eslint-disable react/jsx-no-constructed-context-values */
import { Typography } from '@mui/material';
import ErrorOutlineOutlinedIcon from '@mui/icons-material/ErrorOutlineOutlined';
import * as React from 'react';
import {
  QueueBox, QueueText, ControlBar,
} from '../styles';
import { ModMainPage } from '../../ModerationMainPage/styles';
import BanPopUp from './BanUserPopUp/BanPopUp/BanPopUp';
import { RedditButton } from '../../styles';
import BannedUserList from './NonEmptyBanned/BannedUserList/BannedUserList';

export const BanContext = React.createContext();

/**
 * the whole page of banned
 * @component
 * @return {React.Component} - banned user page
 */

function Ban() {
  const [openBan, setOpenBan] = React.useState(false);

  const handleClickOpenBan = () => { setOpenBan(true); };
  const handleClickCloseBan = () => { setOpenBan(false); };
  return (
    <ModMainPage>
      <BanContext.Provider value={{
        openBan, handleClickCloseBan,
      }}
      >
        <BanPopUp />
      </BanContext.Provider>
      <ControlBar>
        <RedditButton
          fontSize="14px"
          padding="0px 24px 0px 24px"
          fontWeight="bold"
          variant="contained"
          onClick={handleClickOpenBan}
        >
          Ban user
        </RedditButton>
      </ControlBar>
      <QueueBox>
        <QueueText>
          <Typography variant="h6">Banned users</Typography>
          <ErrorOutlineOutlinedIcon color="primary" />
        </QueueText>
      </QueueBox>
      <BannedUserList />
    </ModMainPage>
  );
}

export default Ban;
