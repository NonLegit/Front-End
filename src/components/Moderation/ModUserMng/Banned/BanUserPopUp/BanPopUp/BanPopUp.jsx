/* eslint-disable import/no-cycle */
import * as React from 'react';
import { Divider, Box } from '@mui/material';
import { useParams } from 'react-router-dom';
import { StyledDialog } from '../../../styles';
import { BanContext } from '../../BanUser';
import Header from '../../../Header/Header';
import Username from '../../../Username/Username';
import BanReason from '../BanReason/BanReason';
import BanNote from '../BanNote/BanNote';
import BanPeriod from '../BanPeriod/BanPeriod';
import BanMessage from '../BanMessage/BanMessage';
import BanFooter from '../BanFooter/BanFooter';
import { banUnbanUser } from '../banServer';

/**
 * Ban popup
 * @component
 * @return {React.Component} - Ban popup component
 */

function BanPopUp() {
  const { subReddit } = useParams();
  const {
    openBan, handleClickCloseBan,
  } = React.useContext(BanContext);

  const handleBan = async () => {
    const userName = document.getElementById('username').value;
    const reason = document.getElementById('reason').textContent;
    const banNote = document.getElementById('banNote').value;
    const banMessage = document.getElementById('banMessage').value;
    const duration = document.getElementById('duration').value;
    banUnbanUser(userName, subReddit, 'ban', reason, banNote, banMessage, duration);
  };
  return (
    <StyledDialog
      fullScreen
      open={openBan}
      width="538px"
      height="625px"
    >
      <Box>
        <Header buttonFunction={handleClickCloseBan} headerText="Ban a user:" />
        <Divider />
        <Username placeholder="u/username" />
        <BanReason punishType="Spam" />
        <BanNote note="" />
        <BanPeriod duration={-1} />
      </Box>
      <Box sx={{ backgroundColor: '#edeff1 ', height: '100%' }}>
        <BanMessage />
        <BanFooter handleClickCloseBan={handleClickCloseBan} handleBan={handleBan} />
      </Box>

    </StyledDialog>
  );
}

export default BanPopUp;
