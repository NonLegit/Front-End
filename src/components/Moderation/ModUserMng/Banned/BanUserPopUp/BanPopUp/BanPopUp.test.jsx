import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import BanPopUp from './BanPopUp';
import BanUser from '../../BanUser';

test('test snapshot', async () => {
  const tree = renderer.create(<Router><BanUser><BanPopUp /></BanUser></Router>).toJSON();
  expect(tree).toMatchSnapshot();
});
