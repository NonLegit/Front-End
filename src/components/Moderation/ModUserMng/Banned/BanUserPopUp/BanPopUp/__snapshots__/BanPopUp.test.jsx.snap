// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div
  className="MuiBox-root css-a3vfuy"
>
  <div
    className="MuiBox-root css-1vog110"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1d577ig-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      fontSize="14px"
      fontWeight="bold"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      padding="0px 24px 0px 24px"
      tabIndex={0}
      type="button"
    >
      Ban user
    </button>
  </div>
  <div
    className="MuiBox-root css-uhqeeq"
  >
    <div
      className="MuiBox-root css-o76n3x"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
      >
        Banned users
      </h6>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
        data-testid="ErrorOutlineOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11 15h2v2h-2v-2zm0-8h2v6h-2V7zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
    </div>
  </div>
  <div
    className="MuiBox-root css-ekpinx"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-zbb00y-MuiSvgIcon-root"
      data-testid="NotInterestedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8 0-1.85.63-3.55 1.69-4.9L16.9 18.31C15.55 19.37 13.85 20 12 20zm6.31-3.1L7.1 5.69C8.45 4.63 10.15 4 12 4c4.42 0 8 3.58 8 8 0 1.85-.63 3.55-1.69 4.9z"
      />
    </svg>
    <h6
      className="MuiTypography-root MuiTypography-h6 css-155szis-MuiTypography-root"
    >
      No banned users in r/
    </h6>
  </div>
</div>
`;
